### PR TITLE
Trigger OTP modal onblur for email field

### PR DESCRIPTION
**Pull request by piyushswain**
This PR triggers the Bolt authentication script on the onblur event for the emailField.

### CODE REVIEW

File: app/assets/javascripts/authorize_account.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class AuthorizeAccount {
+  constructor(emailField, pubKey, domain) {
+    this.emailField = emailField;
+    this.pubKey = pubKey;
+    this.domain = domain;
+  }
+
+  async accessRequest() {
+    console.log('Modified')
```
Comment (Naokimi) [inline app/assets/javascripts/authorize_account.js:9]: might want to take this out

File: app/assets/javascripts/authorize_account.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+  }
+
+  async hasBoltAccount(email) {
+    const encodedEmail = encodeURIComponent(email);
+    // Call the endpoint to check if the email already exist on bolt
+    const response = await fetch(this.domain + "/v1/account/exists?email=" + encodedEmail);
+    const responseAsJson = await response.json();
+    return responseAsJson.has_bolt_account
+  }
```
Comment (Naokimi) [inline app/assets/javascripts/authorize_account.js:42]: is this used somewhere?

**Review by DanielePalombo — APPROVED** [review]
LGTM.

Please fix the requested changes from @Naokimi

File: app/assets/javascripts/authorize_account.js (hunk 2 above)
Comment (piyushswain) [reply app/assets/javascripts/authorize_account.js:42]: The `hasBoltAccount` method checks if the email entered exists on Bolt or not. Depending on the response we show the login authentication modal

Comment (piyushswain) [reply app/assets/javascripts/authorize_account.js:42]: Oh I see
Removed this now

Comment (piyushswain) [reply app/assets/javascripts/authorize_account.js]: check here [dc7e3f5](https://github.com/nebulab/bolt_demo/commit/dc7e3f594070c4d9533b805f52ee0da9283907af)

File: app/assets/javascripts/authorize_account.js (hunk 1 above)
Comment (piyushswain) [reply app/assets/javascripts/authorize_account.js]: done

**Pull request merged**